components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsGraphBuilder.java

```diff
@@ -280,19 +286,31 @@ protected void filterOptions(Map<String, Map<String, String>> authenticationOpti
                     }
                     removeOption = true;
                 } else if (!authenticators.isEmpty()) {
-                    // Both idp and authenticator present, but authenticator is given by display name due to the fact
-                    // that it is the one available at UI. Should translate the display name to actual name, and
-                    // keep/remove option
+                    // Both idp and authenticator present.
                     removeOption = true;
 
                     if (FrameworkConstants.LOCAL_IDP_NAME.equals(idpName)) {
                         List<LocalAuthenticatorConfig> localAuthenticators = ApplicationAuthenticatorService
                             .getInstance().getLocalAuthenticators();
                         for (LocalAuthenticatorConfig localAuthenticatorConfig : localAuthenticators) {
-                            if (authenticatorConfig.getName().equals(localAuthenticatorConfig.getName()) &&
-                                authenticators.contains(localAuthenticatorConfig.getDisplayName().toLowerCase())) {
-                                removeOption = false;
-                                break;
+                            if (StringUtils.isNotBlank(authenticatorNameResolver[0]) &&
+                                    authenticatorNameResolver[0].equals(FrameworkConstants.JSAttributes.
+                                            AUTHENTICATOR_NAME_RESOLVER_AUTHENTICATOR_NAME_VALUE)) {
+                                if (authenticatorConfig.getName().equals(localAuthenticatorConfig.getName()) &&
+                                        authenticators.contains(localAuthenticatorConfig.getName())) {
+                                    removeOption = false;
+                                    break;
+                                }
+                            } else {
+                                // If the authenticatorNameResolver is not defined or is equal to the authenticator
+                                // display name, authenticator display name will be used to resolve the
+                                // authentication options.
+                                // Should translate the display name to actual name, and keep/remove option.
+                                if (authenticatorConfig.getName().equals(localAuthenticatorConfig.getName()) &&
+                                        authenticators.contains(localAuthenticatorConfig.getDisplayName())) {
```

Comment:
`toLowerCase()` is removed here in favor of autheticator name.
i.e. `filteredOptions.get(idp).add(authenticator.toLowerCase());` need to remove authenticator being lower case here in favor of authenticator name.

Reply:
wouldn't this affect existing scripts if the authenticator display name used is case insensitive

Reply:
Fixed by first resolving the `authenticatorNameResolver` prior to looping through list to check authenticators